edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/Handlers/GenericSubscriptions.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System.Collections.Generic;
+
+    public class GenericSubscriptions : ISubscriber
```

Comment:
How is this used?

Reply:
The purpose of this class was originally that because more handlers were interested in "$edgehub/+/subscriptions", to avoid repeated subscriptions, there were to options:
- the framework that collects subscriptions, after collecting all, it eliminates the duplications
- move the subscriptions that more handlers interested are into a separate class.

this GenericSubscriptions is a solution of the second option. Since then the handler of "$edgehub/+/subscriptions" was moved into a single class (instead of different handlers handling their of subscription changes), so maybe this class can be removed (and the "$edgehub/+/subscriptions" be moved into that singe class, I will check this option)

Reply:
Made an attempt to remove the GenericSubscription, however it didn't work:
- The handler class for the topics the GenericSubscription subscribes is the 'SubscriptionChangeHandler'
- The 'SubscriptionChangeHandler' is not a normal handler class. It was created because there were three different handler classes handling their own subscription changes, but the related code was almost the same in all 3 classes. That code was moved out into 'SubscriptionChangeHandler'
- The 'SubscriptionChangeHandler' needs to know about other edgehub-bridge components (because it needs to find whose subscriptions it needs to handle), so it has a dependency to ComponentDiscovery.
- Because of that dependency, it cannot be 'discovered' by ComponentDiscovery, (that would be a circular dependency)
- Because if this, SubscriptionChangeHandler is injected directly into MqttBrokerConnector, and the connector uses it apart from the rest of the handlers (it asks SubscriptionChangeHandler first, then the rest of the handlers)

because all of that, SubscriptionChangeHandler cannot just implement ISubscriber listing its subscriptions, so I need to keep a separate class for that. For the same reason I will not be able to merge ISubscriber and IMessageConsumer.

I keep thinking if there is a better way to implement these interfaces

Reply:
With a recent bigger change, managed to remove GenericSubcriptions